### PR TITLE
SEQNG-881 Keep track of guide configuration set by TCC.

### PR DESCRIPTION
**Pull request by jluhrs**
Seqexec now keeps track of the guide configuration set from TCC. It defines a HTTP API, that TCC uses to inform Seqexec when a new configuration is set.

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/gems/GemsController.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server.gems
+
+import seqexec.server.tcs.Gaos.Reason
+import squants.Time
+
+trait GemsController[F[_]] {
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/gems/GemsController.scala]: This is a placeholder until we implement GeMS

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/GuideConfigDb.scala [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                  } yield Lgs(strapLoop, sfoLoop)
+                }
+              }
+            }
+            case _     => Left(DecodingFailure("AltairConfig", c.history))
+          }
+        }
+        else Right(AltairOff)
+      }
+  }
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/GuideConfigDb.scala]: I am sure there is a prettier way to do this.

**Comment by cquiroz** [comment]
in general looks very good. Using the `Ref` to store the guide config should be very simple and attaching it to the http4s server looks great

File: modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala [hunk 3]
```diff
@@ -158,7 +162,8 @@ object WebServerLauncher extends IOApp with LogInitialization with SeqexecConfig
       "/"                     -> new StaticRoutes(conf.devMode, OcsBuildInfo.builtAtMillis, bec).service,
       "/"                     -> PrometheusExportService[IO](cr).routes,
       "/api/seqexec/commands" -> new SeqexecCommandRoutes(as, inputs, se).service,
-      "/api"                  -> new SeqexecUIApiRoutes(conf.site, conf.devMode, as, outputs).service
+      "/api"                  -> new SeqexecUIApiRoutes(conf.site, conf.devMode, as, outputs).service,
+      "/api/seqexec/guide"    -> new GuideConfigDbRoutes(gcdb).service
```
[the remark this answers is not in the document]
Comment (jluhrs) [reply modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala]: I did test it with the TCC

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/GuideConfigDb.scala [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      if(_) for {
+        srcs <- c.downField("sources").as[Set[TipTiltSource]]
+        coma <- c.downField("comaOn").as[ComaOption]
+      } yield M2GuideOn(coma, srcs)
+      else Right(M2GuideOff)
+    }
+  }
+
+  implicit val tcsGuideConfigDecoder: Decoder[TcsGuideConfig] =
+    Decoder.forProduct3("mountGuideOn", "m1Guide", "m2Guide")(TcsGuideConfig)
```
[the remark this answers is not in the document]
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/tcs/GuideConfigDb.scala]: It's a helper constructor. `forProduct3[A, B, C, T](String, String, String)((A, B, C) => T)`
It reads the 3 fields from the cursor as the implicit types A, B and C, and uses the second parameter to build a T

**Pull request merged**